File: app/src/free/java/com/example/dagger/MainActivityModule.java
package com.example.dagger;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by LKW on 24/6/2017.
 */

@Module
public abstract class MainActivityModule {
    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();
}

File: app/src/free/java/com/example/dagger/MainActivityModule.kt
package com.example.dagger

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by LKW on 24/6/2017.
 */

@Module
abstract class MainActivityModule {
    @ContributesAndroidInjector
    internal abstract fun mainActivity(): MainActivity
}
